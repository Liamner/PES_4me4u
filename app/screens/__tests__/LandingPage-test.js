import 'react-native';
import React from 'react';
import Login from '../LoginScreen';
import renderer from 'react-test-renderer'
import {render, fireEvent} from "@testing-library/react-native"

let component

test('LandingPage snapshot', ()=>{
    const snap = renderer.create(
        <Login/>
    ).toJSON();
    expect(snap).toMatchSnapshot();
});

describe('<Login/>', ()=> {
    //antes de cada prueba creamos un componente "limpio"
    beforeEach(() => {
        component = render(<Login/>);
    });

    it('renders correctly v1', () =>{
        expect(component).toBeDefined();
        
        //expect(component.getByTestId('test2')).toBeDefined(); //se puede poner ids de test a los componentes
    });

    it('renders correctly v2', ()=>{
        const tree = renderer
        .create(<Login/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
     
});