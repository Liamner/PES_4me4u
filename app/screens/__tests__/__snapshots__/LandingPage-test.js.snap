// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> renders correctly v2 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 373.52000000000004,
          "width": 373.52000000000004,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Presta, Intercambia y Regala
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4283197087,
              4282079640,
              4279840618,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
            Continuar con Facebook
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294967295,
              4294967295,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "width": 200,
            },
            Object {
              "borderColor": "#a2cff0",
              "borderWidth": 1,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "color": "#000",
            }
          }
        >
            Continuar con Google
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4292412457,
              4293862204,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "material",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
            Continuar con Email
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;

exports[`LandingPage snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 373.52000000000004,
          "width": 373.52000000000004,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Presta, Intercambia y Regala
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4283197087,
              4282079640,
              4279840618,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
            Continuar con Facebook
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294967295,
              4294967295,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "width": 200,
            },
            Object {
              "borderColor": "#a2cff0",
              "borderWidth": 1,
            },
          ]
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#000",
            }
          }
        >
            Continuar con Google
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4292412457,
              4293862204,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
            Continuar con Email
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;
